File: sample/flutter_base/packges/common_network/lib/src/CommonNetwork.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:service_interface/service_interface.dart';

class CommonNetwork implements NetworkProtocol  {
  void onLoad() { }

  void onUnLoad() { }

  void post(Uri uri, Map params, void Function(String? body, Object? error) callback) {
    final headers = {'Content-Type': 'application/json'};
    final body = jsonEncode(params);
    http.post(uri, headers: headers, body: body, encoding: utf8).then((final response) {
      if (response.statusCode >= 200 && response.statusCode <= 299) {
        callback(response.body, null);
      } else {
        callback(null, response.body);
      }
    }).catchError((Object error) {
      callback(null, error);
    });
  }
}

File: output/src/services/CommonNetwork.ts
import { NetworkProtocol } from './service_interface';

export class CommonNetwork implements NetworkProtocol {
  onLoad(): void {
    // no-op
  }

  onUnLoad(): void {
    // no-op
  }

  post(uri: string, params: Record<string, any>, callback: (body: string | null, error: object | null) => void): void {
    const headers = { 'Content-Type': 'application/json' };
    const body = JSON.stringify(params);

    fetch(uri, {
      method: 'POST',
      headers,
      body,
    })
      .then((response) => {
        if (response.ok) {
          response.text().then((text) => callback(text, null));
        } else {
          response.text().then((text) => callback(null, new Error(text)));
        }
      })
      .catch((error: unknown) => {
        callback(null, error as object | null);
      });
  }
}